File: src/main/java/dev/zprestige/fire/module/client/panel/TickListener.java
package dev.zprestige.fire.module.client.panel;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.TickEvent;
import dev.zprestige.fire.ui.menu.panel.PanelScreen;

public class TickListener extends EventListener<TickEvent, Panel> {

    public TickListener(final Panel panel) {
        super(TickEvent.class, panel);
    }

    @Override
    public void invoke(final Object object) {
        if (!(mc.currentScreen instanceof PanelScreen)) {
            module.disableModule();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/sprint/TickListener.kt
package dev.zprestige.fire.module.movement.sprint

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.PacketEvent
import dev.zprestige.fire.util.impl.EntityUtil

class TickListener(sprint: Sprint) :
    EventListener<PacketEvent.PacketReceiveEvent, Sprint>(PacketEvent.PacketReceiveEvent::class.java, sprint) {

    override fun invoke(e: Any) {
        if (module.nullCheck() && EntityUtil.isMoving()){
            mc.player.isSprinting = true
        }
    }
}